teralaunch/src/utils.js

```
@@ -0,0 +1,127 @@
+ * @param {Array<number>} speedHistory - The history of download speeds.
+ * @param {number} speedHistoryMaxLength - The maximum length of the speed history.
+ * @returns {number} the estimated time remaining in seconds, or 0 if the input is invalid. The result is capped at 30 days maximum.
+ */
+export function calculateGlobalTimeRemaining(totalDownloadedBytes, totalSize, speed, speedHistory, speedHistoryMaxLength) {
+  console.log("Calculating global time remaining:", {
+    totalDownloadedBytes,
+    totalSize,
+    speed,
+  });
+  if (
+    !isFinite(speed) ||
```

Biome (1.9.4)

[error] isFinite is unsafe. It attempts a type coercion. Use Number.isFinite instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isFinite instead.


(lint/suspicious/noGlobalIsFinite)